tests/eoc_test.cpp

```diff
@@ -0,0 +1,46 @@
+#include <map>
+#include <memory>
+#include <set>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "avatar.h"
+#include "calendar.h"
+#include "cata_catch.h"
+#include "character.h"
+#include "common_types.h"
+#include "creature_tracker.h"
+#include "effect_on_condition.h"
+#include "faction.h"
+#include "field.h"
+#include "field_type.h"
+#include "game.h"
+#include "line.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "memory_fast.h"
+#include "npc.h"
+#include "npc_class.h"
+#include "optional.h"
+#include "overmapbuffer.h"
+#include "pimpl.h"
+#include "player_helpers.h"
+#include "point.h"
+#include "text_snippets.h"
+#include "type_id.h"
+#include "units.h"
+
+static const effect_on_condition_id effect_on_condition_EOC_teleport_test( "EOC_teleport_test" );
+
+
+TEST_CASE( "EOC_teleport", "[eoc]" )
+{
+    tripoint_abs_ms before = get_avatar().get_location();
```

Comment:
The rest looks good to me, but it looks like there's a problem in the new test:
https://github.com/CleverRaven/Cataclysm-DDA/runs/6971512465?check_suite_focus=true#step:16:401

![test_fail](https://user-images.githubusercontent.com/12537966/175874080-8c96b198-2667-47e4-83eb-1fa42cca1079.png)

I think there's a chance for an obstacle to spawn on the map during the test. You could probably fix it by clearing the map:
```suggestion
    clear_map();
    tripoint_abs_ms before = get_avatar().get_location();
```